lib/msal-core/src/utils/UrlUtils.ts

```diff
@@ -244,4 +245,27 @@ export class UrlUtils {
         const hash = UrlUtils.getHashFromUrl(urlFragment);
         return CryptoUtils.deserialize(hash);
     }
+
+    /**
+     * @hidden
+     * generate unique state per request
+     * @param request
+     */
+    static getRequestState(state: string): string {
+        // append GUID to user set state  or set one for the user if null
+        return state && !StringUtils.isEmpty(state) ? CryptoUtils.createNewGuid() + "|" + state : CryptoUtils.createNewGuid();
+    }
+
+    /**
+     * @hidden
+     * validate correlationId and generate if not valid or not set by the user
+     * @param correlationId
+     */
+    static getRequestCorrelationId(correlationId: string): string {
```

Comment:
Same comment as above for getRequestState

Reply:
Following this up with a requestValidation utility function in a new PR.